clang/lib/Format/Format.cpp

```diff
@@ -1698,8 +1698,6 @@ FormatStyle getGoogleStyle(FormatStyle::LanguageKind Language) {
           /*BasedOnStyle=*/"google",
       },
   };
-  GoogleStyle.AttributeMacros.push_back("GUARDED_BY");
```

Comment:
Even if that would not be needed anymore to achieve the expected formatting, these are AttributeMacros, and should be declared as such.

Reply:
They are attribute macros indeed, the problem is that we actually need more. The ones used with fields are at least:
- ABSL_PT_GUARDED_BY
- ABSL_ACQUIRED_AFTER
- ABSL_ACQUIRED_BEFORE
- ABSL_GUARDED_BY_FIXME

We could also consider including the annotations for functions, but the patch only broke formatting for variables, so it's not strictly necessary to unblock the release.

If we want to also include the ones that are used with functions (they are not strictly necessary because `clang-format` does a decent job there without config), we would need to add at least these:
- ABSL_EXCLUSIVE_LOCKS_REQUIRED
- ABSL_LOCKS_EXCLUDED
- ABSL_LOCK_RETURNED
- ABSL_EXCLUSIVE_LOCK_FUNCTION
- ABSL_EXCLUSIVE_TRYLOCK_FUNCTION
- ABSL_SHARED_TRYLOCK_FUNCTION
- ABSL_ASSERT_EXCLUSIVE_LOCK
- ABSL_ASSERT_SHARED_LOCK
- ABSL_NO_THREAD_SAFETY_ANALYSIS


I am not sure how to best approach it and would appreciate some guidance here. Should we have all these attribute macros inside `AttributeMacros` or should we aim for clang-format formatting them reasonably without configuration?

Reply:
@HazardyKnusperkeks friendly ping. Any thoughts on including a few more attributes into the (the first list of 4 elements) vs landing this change and relying on implicit formatting of those as function names?

I am happy to choose one of the two options arbitrarily, but I don't have enough context on `clang-format` to understand which approach is preferable, so I would love to get an opinion from someone in the `clang-format` community.

Reply:
FWIW, not recognizing `foo` in `int foo BAR` as start-of-name looks like a big enough regression (which seems to be the main reason behind the line-braking behavior change), independent of whatever we do with the list of attribute-macros, I believe we should still make sure annotations for `foo` are correct rather urgently. so I am actually still in favor of landing this patch as-is, rather than trying to fix final formatting in a bunch of special cases by updating `AtrributeMacros` list.

Reply:
I'm open in all directions.

When clang-format does format attribute macros out of the box correctly, that is nice. But I wouldn't put (too much) work into it, if declaring them to clang-format as what they are fixes all misformatting.

Thus I would keep the entries in `AttributeMacros`.

Reply:
I think we should postpone the inclusion of those names into `AtrributeMacros`  and land the patch as is.

It seems that there is agreement among everyone that having this formatting without explicit `AtrributeMacros` is desirable, so landing as is is a no-brainer.

Whether we should add common macro names into `AttributeMacros` is more contentious, so I think we may need more data to back up our decision to go either way. I have some examples where not having those names in the config leads to undesirable formatting, but I would share them in a follow-up conversation.